command/agent/config/config.go

```diff
@@ -1112,9 +1113,9 @@ func parseExec(result *Config, list *ast.ObjectList) error {
 		return err
 	}
 
-	// if user does not specify a restart signal, use a default
+	// if the user does not specify a restart signal, default to SIGTERM
 	if ec.RestartStopSignal == nil {
-		ec.RestartStopSignal = os.Interrupt
+		ec.RestartStopSignal = syscall.SIGTERM
```

Comment:
using `os.Interrupt` was intentional here

in this case, it maps to syscall.SIGINT

Reply:
IIRC, SIGTERM is a more conventional default for soft exit

Reply:
If [this stackoverflow](https://stackoverflow.com/a/4047975) is to be believed:
> SIGTERM and SIGKILL are intended for general purpose "terminate this process" requests. SIGTERM (by default) and SIGKILL (always) will cause process termination. SIGTERM may be caught by the process (e.g. so that it can do its own cleanup if it wants to), or even ignored completely; but SIGKILL cannot be caught or ignored.
>
> SIGINT and SIGQUIT are intended specifically for requests from the terminal: particular input characters can be assigned to generate these signals (depending on the terminal control settings).